### PR TITLE
KAFKA-6567: Remove KStreamWindowReducer

### PR DESCRIPTION
**Pull request by samuel-hawker**
This pull request removes the final reference to `KStreamWindowReducer` and replaces it with `KStreamWindowAggregate`

Signed-off-by: Samuel Hawker <sam.b.hawker@gmail.com>

contribution is my original work and that I license the work to the project under the project's open source license.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/TimeWindowedKStreamImpl.java [hunk 1]
```diff
@@ -135,6 +135,7 @@
         Objects.requireNonNull(materialized, "materialized can't be null");
 
         final MaterializedInternal<K, V, WindowStore<Bytes, byte[]>> materializedInternal = new MaterializedInternal<>(materialized);
+        final Aggregator<K, V, V> reduceAggregator = aggregatorForReducer(reducer);
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/TimeWindowedKStreamImpl.java]: Nit: I would inline this method. It's a single liner and only used once.

**Review by mjsax — COMMENTED** [review]
Just one nit. Overall LGTM.

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/TimeWindowedKStreamImpl.java (hunk 1 above)
Comment (samuel-hawker) [reply streams/src/main/java/org/apache/kafka/streams/kstream/internals/TimeWindowedKStreamImpl.java]: I have inlined it, good catch.

**Comment by samuel-hawker** [comment]
The build failed, not sure why, they worked locally and I only moved the function call inline

**Comment by samuel-hawker** [comment]
retest this please

**Review by guozhangwang — APPROVED** [review]
LGTM!

**Comment by mjsax** [comment]
Thanks for the PR @samuel-hawker and congrats to your first contribution to Kafka! 🎉

**Pull request merged**